src/features/query-bar/components/filter/FilterButton.js

```diff
@@ -41,10 +42,11 @@ class FilterButton extends React.Component<Props, State> {
         super(props);
 
         this.state = {
-            transientConditions: this.props.conditions.slice(0),
-            selectedConnector: AND,
             areErrorsEnabled: false,
+            hasAppliedFilters: false,
             isMenuOpen: false,
+            selectedConnector: AND,
+            transientConditions: this.props.conditions.slice(0),
```

Comment:
There's a bug here. It's doing only a shallow copy which falls short of an immutable copy. Should use lodash.cloneDeep. 

I see multiple references to slice under src/features/query-bar//components.